Fatras/include/ActsFatras/Utilities/GammaDistribution.hpp

```diff
@@ -0,0 +1,120 @@
+    using distribution_type = GammaDistribution;
+
+    /// Location parameter.
+    ///
+    /// Shape parameter
+    double alpha = 0.5;
+    /// Scale parameter.
+    double theta = 1.0;
+
+    /// Construct from parameters.
+    param_type(double alpha_, double theta_) : alpha(alpha_), theta(theta_) {}
+    // Explicitly defaulted construction and assignment
+    param_type() = default;
+    param_type(const param_type &) = default;
+    param_type(param_type &&) = default;
+    param_type &operator=(const param_type &) = default;
+    param_type &operator=(param_type &&) = default;
+
+    /// Parameters should be EqualityComparable
+    friend bool operator==(const param_type &lhs, const param_type &rhs) {
+      return (lhs.alpha == rhs.alpha) && (lhs.theta == rhs.theta);
+    }
+  };
+  /// The type of the generated values.
+  using result_type = double;
+
+  /// Construct directly from the distribution parameters.
+  GammaDistribution(double alpha, double theta) : m_cfg(alpha, theta) {}
+  /// Construct from a parameter object.
+  explicit GammaDistribution(const param_type &cfg) : m_cfg(cfg) {}
+  // Explicitlely defaulted construction and assignment
+  GammaDistribution() = default;
+  GammaDistribution(const GammaDistribution &) = default;
+  GammaDistribution(GammaDistribution &&) = default;
+  GammaDistribution &operator=(const GammaDistribution &) = default;
+  GammaDistribution &operator=(GammaDistribution &&) = default;
+
+  /// Reset any possible internal state. Noop, since there is no internal state.
+  void reset() {}
+  /// Return the currently configured distribution parameters.
+  param_type param() const { return m_cfg; }
+  /// Set the distribution parameters.
+  void param(const param_type &cfg) { m_cfg = cfg; }
+
+  /// The minimum value the distribution generates.
+  result_type min() const { return 0.; }
+  /// The maximum value the distribution generates.
+  result_type max() const { return std::numeric_limits<double>::infinity(); }
+
+  /// Generate a random number from the configured gamma distribution.
+  template <typename Generator>
+  result_type operator()(Generator &generator) {
+    return (*this)(generator, m_cfg);
+  }
+  /// Generate a random number from the given gamma distribution.
+  template <typename Generator>
+  result_type operator()(Generator &generator, const param_type &params) {
+    if (params.alpha >= 1.) {
+      std::gamma_distribution<double> gDist(params.alpha, params.theta);
+      return gDist(generator);
+    } else if (params.alpha <= 0.) {
+      return 0.;
+    } else {
+      std::gamma_distribution<double> gDistPlus1(params.alpha + 1.,
+                                                 params.theta);
+      // Get a random number using alpha+1
+      const auto uPlus1 = gDistPlus1(generator);
+      std::uniform_real_distribution<double> uDist(0., 1.);
+      double u2 = 0.;
+      do {
+        u2 = uDist(generator);
+      } while (u2 == 0.);
```

Comment:
```suggestion
      // draw from uniform (0,1)
      std::uniform_real_distribution<double> uDist(std::numeric_limits<double>::min(), 1.);
      double u2 = uDist(generator);
```
would this also do it?

Reply:
There's a small chance that `u2` will be zero. However, it's possible to avoid initialization with 0, and initialize with a random number and then use `while(...) {...}`